Helm scaffold chart template errors

An OOTB scaffold chart (`helm create`) has templating errors as follows:

```console
$ helm install mychart
Error: YAML parse error on mychart/templates/deployment.yaml: error converting YAML to JSON: yaml: line 8: mapping values are not allowed in this context

$ helm lint mychart/
==> Linting mychart/
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/deployment.yaml: unable to parse YAML
	error converting YAML to JSON: yaml: line 8: mapping values are not allowed in this context
[ERROR] templates/service.yaml: unable to parse YAML
	error converting YAML to JSON: yaml: line 8: mapping values are not allowed in this context
[ERROR] templates/tests/test-connection.yaml: unable to parse YAML
	error converting YAML to JSON: yaml: line 8: mapping values are not allowed in this context

Error: 1 chart(s) linted, 1 chart(s) failed
```

Output of `helm version`:
From master

Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.):

--- a/pkg/chartutil/create.go
+++ b/pkg/chartutil/create.go
@@ -301,9 +301,9 @@ Common labels
 app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
 helm.sh/chart: {{ include "<CHARTNAME>.chart" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
-{{- if .Chart.AppVersion -}}
+{{- if .Chart.AppVersion }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
-{{- end -}}
+{{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
 {{- end -}}
 `

